### ISSUE TITLE
Deprecated service call not caught

### ISSUE THREAD

**Issue by danepowell** [mglaman/phpstan-drupal#116]
If this issue should be moved to some other project, please let me know, I'm not exactly sure where these rules live.

I'm using phpstan-drupal-deprecations to scan Acquia Connector in preparation for Drupal 9. We call [`\Drupal::service('path.alias_manager')`](https://git.drupalcode.org/project/acquia_connector/blob/8.x-2.0-beta3/src/Controller/SpiController.php#L413), which is [deprecated in Drupal 8.8 and removed in Drupal 9](https://www.drupal.org/node/3092086).

I would expect phpstan to catch this, but it didn't. Any idea why?

**Comment by mglaman** [comment]
I think it may be the same reason PhpStorm's analysis doesn't work on `::service()->method` but `::getContainer()->get` works.

**Comment by mglaman** [comment]
Actually, there is more to it.

```
  path.alias_manager:
    class: Drupal\Core\Path\AliasManager
    arguments: ['@path_alias.repository', '@path_alias.whitelist', '@language_manager', '@cache.data']
    deprecated: 'The "%service_id%" service is deprecated. Use "path_alias.manager" instead. See https://drupal.org/node/3092086'
```

\PHPStan\Drupal\DrupalServiceDefinition needs to have a deprecated flag, based on the services definition and use its messaging. From there it needs to bubble up a deprecation notice, somehow.

**Comment by mglaman** [comment]
In https://github.com/mglaman/phpstan-drupal/issues/205 I fixed calls to `->get()` on services. This may have been fixed. I didn't add a test fixture for `\Drupal::service` directly, though.

**Comment by mglaman** [comment]
Fixed by https://github.com/mglaman/phpstan-drupal/issues/190
